lib/Actions.js

```diff
@@ -149,9 +245,46 @@ Actions.prototype = {
     },
 
     /**
+     * Wait for an element is on the page and visible
+     * @param {string} selector
+     * @param {number} [timeout]
+     * @param {number} [interval]
+     * @returns {Promise}
+     */
+    waitElementIsVisible: function (selector, timeout, interval) {
+        debug('._waitElementIsVisible() ' + selector);
+        return this.wait(/* @covignore */ function (selector) {
+            var nodes = Array.prototype.slice.call(Sizzle(selector), 0);
+            return nodes.some(function (node) {
+                return node.offsetWidth !== 0 && node.offsetHeight !== 0;
+            });
+        }, function (visible) {
+            return visible;
+        }, [selector], timeout, interval);
+    },
+
+    /**
+     * Wait for an element'c content matches pattern
+     * @param {string} selector
+     * @param {string} pattern
+     * @param {number} [timeout]
+     * @param {number} [interval]
+     * @returns {Promise}
+     */
+    waitForPattern: function (selector, pattern, timeout, interval) {
+        debug('._waitForPattern() %s on selector %s', pattern, selector);
+        return this.wait(/* @covignore */ function (selector) {
+            var nodes = Sizzle(selector);
+            return nodes.length && nodes[0].textContent || '';
```

Comment:
Keep in mind new feature about `attr`. That is not only one way to get info from node for now.